code/control/DeployDispatcher.php

```diff
@@ -124,9 +124,15 @@ public function show(SS_HTTPRequest $request) {
 	 * @return \SS_HTTPResponse
 	 */
 	public function log(SS_HTTPRequest $request) {
-		$deployment = DNDeployment::get()->byId($request->params('ID'));
-		if (!$deployment || !$deployment->exists) {
-			return $this->getAPIResponse(['message' => 'This deployment does not exist'], 404);
+		$deployID = $request->param('ID');
+		$deployment = DNDeployment::get()->byId($deployID);
```

Comment:
Can you copy the way show action does this so it's consistent? just copy/paste it including the 404 response